config/src/main/java/org/springframework/security/config/annotation/web/messaging/AuthorizationManagerMessageMatcherRegistry.java

```diff
@@ -0,0 +1,354 @@
+	 * the Matcher returns false.
+	 * @param patterns the patterns to create
+	 * {@link org.springframework.security.messaging.util.matcher.SimpDestinationMessageMatcher}
+	 * from.
+	 */
+	public Constraint simpMessageDestMatchers(String... patterns) {
+		return simpDestMatchers(SimpMessageType.MESSAGE, patterns);
+	}
+
+	/**
+	 * Maps a {@link List} of {@link SimpDestinationMessageMatcher} instances that match
+	 * on {@code SimpMessageType.SUBSCRIBE}. If no destination is found on the Message,
+	 * then the Matcher returns false.
+	 * @param patterns the patterns to create
+	 * {@link org.springframework.security.messaging.util.matcher.SimpDestinationMessageMatcher}
+	 * from.
+	 */
+	public Constraint simpSubscribeDestMatchers(String... patterns) {
+		return simpDestMatchers(SimpMessageType.SUBSCRIBE, patterns);
+	}
+
+	/**
+	 * Maps a {@link List} of {@link SimpDestinationMessageMatcher} instances. If no
+	 * destination is found on the Message, then the Matcher returns false.
+	 * @param type the {@link SimpMessageType} to match on. If null, the
+	 * {@link SimpMessageType} is not considered for matching.
+	 * @param patterns the patterns to create
+	 * {@link org.springframework.security.messaging.util.matcher.SimpDestinationMessageMatcher}
+	 * from.
+	 * @return the {@link Constraint} that is associated to the {@link MessageMatcher}
+	 */
+	private Constraint simpDestMatchers(SimpMessageType type, String... patterns) {
+		List<MessageMatcher<?>> matchers = new ArrayList<>(patterns.length);
+		for (String pattern : patterns) {
+			Supplier<MessageMatcher<Object>> supplier = new PathMatcherMessageMatcherBuilder(pattern, type);
+			MessageMatcher<?> matcher = new SupplierMessageMatcher(supplier);
+			matchers.add(matcher);
+		}
+		return new Constraint(matchers);
+	}
+
+	/**
+	 * The {@link PathMatcher} to be used with the
+	 * {@link MessageSecurityMetadataSourceRegistry#simpDestMatchers(String...)}. The
+	 * default is to use the default constructor of {@link AntPathMatcher}.
+	 * @param pathMatcher the {@link PathMatcher} to use. Cannot be null.
+	 * @return the {@link MessageSecurityMetadataSourceRegistry} for further
+	 * customization.
+	 */
+	public AuthorizationManagerMessageMatcherRegistry simpDestPathMatcher(PathMatcher pathMatcher) {
+		Assert.notNull(pathMatcher, "pathMatcher cannot be null");
+		this.pathMatcher = pathMatcher;
+		return this;
+	}
+
+	/**
+	 * Maps a {@link List} of {@link MessageMatcher} instances to a security expression.
+	 * @param matchers the {@link MessageMatcher} instances to map.
+	 * @return The {@link Constraint} that is associated to the {@link MessageMatcher}
+	 * instances
+	 */
+	public Constraint matchers(MessageMatcher<?>... matchers) {
+		List<MessageMatcher<?>> builders = new ArrayList<>(matchers.length);
+		for (MessageMatcher<?> matcher : matchers) {
+			builders.add(matcher);
+		}
+		return new Constraint(builders);
+	}
+
+	public AuthorizationManager<Message<?>> build() {
```

Comment:
I think this is more of a builder than a registry since it exposes a build method. It makes me wonder if we are building an AuthorizationManager if we out to just provide a builder on MessageMatcherDelegatingAuthorizationManager as you mentioned. We'd need to double check for package tangles though.